### PR TITLE
Add some test for FilterManager

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: src/filter_manager.coffee [hunk 1]
```diff
@@ -35,6 +37,10 @@ class FilterManager
     createOrUpdate: (config, callback) ->
         log.debug "createOrUpdate"
 
+        unless config.file and config.folder and config.contact \
+                and config.calendar and config.notification
```
Comment (nono) [inline src/filter_manager.coffee]: I think the condition is false. It should be `or` instead of `and`, no?

File: test/unit/filter_manager.coffee [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        filterManager.getFilterName().should.equal \
+            'filter-local-device-config/config'
+
+    it "returns an error if the configuration is empty", (done) ->
+
+        filterManager.createOrUpdate {}, (err) ->
+            err.message.should.not.be.empty()
+            done()
+
+    it "returns an error if the configuration is empty", (done) ->
```
Comment (nono) [inline test/unit/filter_manager.coffee]: The configuration is not empty here, right?

Comment (kosssi) [inline test/unit/filter_manager.coffee]: done

File: src/filter_manager.coffee (hunk 1 above)
Comment (kosssi) [inline src/filter_manager.coffee]: yes ...

done

**Comment by aenario** [comment]
Consider adding https://www.npmjs.com/package/coffee-jshint with the node and undef options : it helps catch a lot of errors (not necessary if we go es6)

File: test/unit/filter_manager.coffee [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        @filterManager.getFilterName().should.equal \
+            'filter-local-device-config/config'
+
+    it "returns an error if the configuration is empty", (done) ->
+
+        @filterManager.createOrUpdate {}, (err) =>
+            err.message.should.not.be.empty()
+            should(err).be.null
+            @filterManager.http.put.called.should.not.be.true()
```
Comment (nono) [inline test/unit/filter_manager.coffee:30]: I prefer `should.be.false()` to `should.not.be.true()`

**Pull request merged**